synapse/config/emailconfig.py

```diff
@@ -237,43 +229,28 @@ def read_config(self, config, **kwargs):
                 "add_threepid_template_success_html", "add_threepid_success.html"
             )
 
-            # Check templates exist
-            for f in [
-                self.email_password_reset_template_html,
-                self.email_password_reset_template_text,
-                self.email_registration_template_html,
-                self.email_registration_template_text,
-                self.email_add_threepid_template_html,
-                self.email_add_threepid_template_text,
-                self.email_password_reset_template_failure_html,
-                self.email_registration_template_failure_html,
-                self.email_add_threepid_template_failure_html,
-                email_password_reset_template_success_html,
-                email_registration_template_success_html,
-                email_add_threepid_template_success_html,
-            ]:
-                p = os.path.join(self.email_template_dir, f)
-                if not os.path.isfile(p):
-                    raise ConfigError("Unable to find template file %s" % (p,))
-
-            # Retrieve content of web templates
-            filepath = os.path.join(
-                self.email_template_dir, email_password_reset_template_success_html
-            )
-            self.email_password_reset_template_success_html = self.read_file(
-                filepath, "email.password_reset_template_success_html"
+            # Read and render web templates
+            (
+                password_reset_template_success_html_template,
+                registration_template_success_html_template,
+                add_threepid_template_success_html_template,
+            ) = self.read_templates(
+                [
+                    email_password_reset_template_success_html,
+                    email_registration_template_success_html,
+                    email_add_threepid_template_success_html,
+                ],
+                self.email_template_dir,
             )
-            filepath = os.path.join(
-                self.email_template_dir, email_registration_template_success_html
-            )
-            self.email_registration_template_success_html_content = self.read_file(
-                filepath, "email.registration_template_success_html"
+
+            self.email_password_reset_template_success_html = (
+                password_reset_template_success_html_template.render()
             )
-            filepath = os.path.join(
-                self.email_template_dir, email_add_threepid_template_success_html
+            self.email_registration_template_success_html_content = (
+                registration_template_success_html_template.render()
```

Comment:
Did these used to not be jinja templates?

Reply:
Yes, and I should've mentioned this somewhere. I'm treating all templates as jinja templates as per https://github.com/matrix-org/synapse/issues/7280

Reply:
I'm a fan. 👍 Just making sure it was done on purpose!